lib/rubocop/cop/lint/useless_ruby2_keywords.rb

```diff
@@ -98,14 +98,22 @@ def inspect_sym(node, sym_node)
           return unless node.parent
 
           method_name = sym_node.value
-          definition = node.parent.each_child_node.detect { |n| method_definition(n, method_name) }
+          definition = find_method_definition(node, method_name)
 
           return unless definition
           return if allowed_arguments(definition.arguments)
 
           add_offense(node, message: format(MSG, method_name: method_name))
         end
 
+        def find_method_definition(node, method_name)
+          node.each_ancestor.lazy.map do |ancestor|
```

Comment:
Because we still support Ruby 2.6 for windows `filter_map` is not available. See https://github.com/rubocop/rubocop/actions/runs/3929998645/jobs/6719462639

With this we could refactor to:

```ruby
def find_method_definition(node, method_name)
  node.each_ancestor.lazy.filter_map do |ancestor|
    ancestor.each_child_node(:def, :block, :numblock).find do |child|
      method_definition(child, method_name)
    end
  end.first
end
```

Reply:
FYI, `filter_map` method has been introduced in Ruby 2.7. So, it's not a Windows platform-specific issue.

Reply:
@koic Yes, you are right and sorry for implying it was the case. It was unintentional. My main question was why we support Ruby 2.6 only for windows? :shrug:

Reply:
Ruby 2.6 runtime also supports non-Windows platforms. CI on other OS would surely have failed on CircleCI.